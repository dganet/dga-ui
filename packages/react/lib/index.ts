import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}