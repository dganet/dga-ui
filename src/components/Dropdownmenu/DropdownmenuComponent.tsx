import React from 'react'

import {
    Container,
    FloatContainerStyled,
    FloatingItemStyled,
    ContentMenu
} from './DropdownmenuComponent.styles'
import {
    DropdownmenuProps,
    FloatingContainerTypes
} from './DropdownmenuComponent.types'
import { IoIosArrowDown, IoIosArrowUp, IoIosCafe } from 'react-icons/io'
import { useTheme } from '../../hooks/useTheme/'

const DropdownmenuComponent = ({
    children,
    floatingSide,
    floatingItems
}: DropdownmenuProps) => {
    const [isOpen, setIsOpen] = React.useState(false)
    const theme = useTheme()
    return (
        <Container>
            <ContentMenu onClick={() => setIsOpen(!isOpen)}>
                {children}
                {isOpen ? (
                    <IoIosArrowUp size={14} />
                ) : (
                    <IoIosArrowDown size={14} />
                )}
            </ContentMenu>
            <FloatingContainer floatingSide={floatingSide} isOpen={isOpen}>
                {floatingItems}
            </FloatingContainer>
        </Container>
    )
}

export const FloatingContainer: React.FC<FloatingContainerTypes> = ({
    floatingSide = 'left',
    isOpen,
    children
}) => {
    const theme = useTheme()
    return (
        <FloatContainerStyled
            floatingSide={floatingSide}
            theme={theme}
            isOpen={isOpen}
        >
            {children}
        </FloatContainerStyled>
    )
}

export const FloatingItem: React.FC = ({ children }) => {
    const theme = useTheme()
    return <FloatingItemStyled theme={theme}>{children}</FloatingItemStyled>
}

export default DropdownmenuComponent
