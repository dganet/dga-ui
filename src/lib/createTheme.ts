import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
