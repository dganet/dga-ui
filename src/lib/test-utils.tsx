import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
