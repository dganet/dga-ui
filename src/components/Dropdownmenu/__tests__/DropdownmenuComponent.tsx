import { render, screen } from '../../../lib/test-utils'

import DropdownmenuComponent from '../DropdownmenuComponent'

describe('<DropdownmenuComponent />', () => {
  it('should render the component', () => {
    const { container } = render(<DropdownmenuComponent />)

    expect(screen.getByRole('heading', { name: /Dropdownmenu/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
