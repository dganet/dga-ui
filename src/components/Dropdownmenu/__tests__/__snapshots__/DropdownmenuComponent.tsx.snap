// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownmenuComponent /> should render the component 1`] = `
<div
  class="sc-bdfBQB"
>
  <h1>
    Dropdownmenu - 
    new component
  </h1>
</div>
`;
